src/genai/credentials.py

```diff
@@ -19,3 +22,10 @@ def __init__(
         if api_endpoint is None:
             raise ValueError("api_endpoint must be provided")
         self.api_endpoint = api_endpoint
+        self._remove_api_endpoint_version()
+
+    def _remove_api_endpoint_version(self) -> None:
+        has_version = re.search(r".\d$", self.api_endpoint)
+        if has_version:
+            self.api_endpoint = self.api_endpoint.rsplit("/v", 1)[0]
+            print("Warning: Api_endpoint should not contain any version, removing it")
```

Comment:
use logger warning, not a basic print (look at `Model` class)

Reply:
I thought the logging isnt displayed to the user?, there is:
`logging.getLogger(__name__).addHandler(logging.NullHandler())`

Reply:
```python
def _remove_api_endpoint_version(self) -> None:
    [api, *version] = re.split(r"(/v\d+$)", self.api_endpoint, maxsplit=1)
    if version:
        warn(
            DeprecationWarning(
                f"The 'api_endpoint' property should not contain any explicit API version (rename it from '{self.api_endpoint}' to just '{api}')"
            )
        )
        self.api_endpoint = api
```

Reply:
DeprecationWarning is ignored by default (https://docs.python.org/3/library/exceptions.html#DeprecationWarning), should I rename it to Warning or is this wanted?